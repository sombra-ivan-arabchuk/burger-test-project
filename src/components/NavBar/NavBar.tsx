import React from 'react';
import { AppBar, Button, makeStyles, Toolbar } from '@material-ui/core';
import Typography from '@material-ui/core/Typography';
import { useAuth } from '../../hooks/useAuth';
import { withRouter } from 'react-router';
import { History } from 'history';

const useStyles = makeStyles(theme => ({
  grow: {
    flexGrow: 1,
  },
  appBar: {
    background: 'rgb(255,141,85)',
  },
  title: {
    display: 'none',
    [theme.breakpoints.up('sm')]: {
      display: 'block',
    },
  },
  sectionDesktop: {
    display: 'none',
    [theme.breakpoints.up('md')]: {
      display: 'flex',
    },
  },
  sectionMobile: {
    display: 'flex',
    [theme.breakpoints.up('md')]: {
      display: 'none',
    },
  },
}));

interface NavBarProps {
  history: History;
}

const NavBar = ({ history }: NavBarProps): React.ReactElement => {
  const classes = useStyles();
  const { signIn, signOut, user } = useAuth();

  const AccessBlock =
    user.name === '' ? (
      <Button color="inherit" onClick={(): Promise<string> => signIn(history)}>
        Login
      </Button>
    ) : (
      <Button color="inherit" onClick={(): void => signOut(history)}>
        Logout
      </Button>
    );

  return (
    <div className={classes.grow}>
      <AppBar position="static" className={classes.appBar}>
        <Toolbar>
          <Typography className={classes.title} variant="h6" noWrap>
            Burger-App
          </Typography>
          <div className={classes.grow} />
          <div className={classes.sectionDesktop}>{AccessBlock}</div>
          <div className={classes.sectionMobile}>{AccessBlock}</div>
        </Toolbar>
      </AppBar>
    </div>
  );
};

export default withRouter(NavBar);
