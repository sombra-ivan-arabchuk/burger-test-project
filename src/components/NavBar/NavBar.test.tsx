import React from 'react';
import { render, RenderResult, waitForElement } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import NavBar from './NavBar';
import { Router } from 'react-router';
import { createMemoryHistory } from 'history';

export interface SetupProps {
  container: HTMLElement;
  utils: RenderResult;
}

jest.mock('../../hooks', () => ({
  useAuth: (): object => ({ user: { name: 'test' } }),
}));

const setup = (): SetupProps => {
  // eslint-disable-next-line no-undef
  const history = createMemoryHistory();
  const utils = render(
    <Router history={history}>
      <NavBar />
    </Router>,
  );
  return {
    container: utils.container,
    utils,
  };
};

describe('Catalog', () => {
  it('check if Login button exists', async () => {
    const { utils } = setup();
    const { getAllByText } = utils;
    await waitForElement(() => getAllByText('Login'));
  });
});
