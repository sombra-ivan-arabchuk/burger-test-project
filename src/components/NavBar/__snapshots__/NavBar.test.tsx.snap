// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalog renders 1`] = `
<div
  class="makeStyles-grow-1"
>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic makeStyles-appBar-2 MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        class="MuiTypography-root makeStyles-title-3 MuiTypography-h6 MuiTypography-noWrap"
      >
        Burger-App
      </h6>
      <div
        class="makeStyles-grow-1"
      />
      <div
        class="makeStyles-sectionDesktop-4"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Login
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="makeStyles-sectionMobile-5"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Login
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </header>
</div>
`;
